awpy/analytics/map_control.py

```diff
@@ -1,330 +1,368 @@
 
     Raises:
         ValueError: If map_name is not in awpy.data.NAV
-                    If current_player_data not of expected
-                    awpy frame format
+
+    Values are allocated to tiles depending on how many tiles are between it
+    and the source tile (aka tile distance). The smaller the tile distance,
+    the close the tile's value is to 1. The tile distance can get as high as
+    the max_depth argument, which is 10 as default. If the max_depth is kept
+    as default 10, this means the BFS search will stop once a tile
+    distance of 10 is reached.
     """
     if map_name not in NAV:
-        raise ValueError("Map not found.")
-
-    player_positions = extract_player_positions(frame)
+        msg = "Map not found."
+        raise ValueError(msg)
 
-    return _calc_map_control_helper(
+    return calc_parsed_frame_map_control_values(
         map_name=map_name,
-        current_player_data=player_positions,
+        current_player_data=extract_teams_metadata(frame),
     )
 
 
-def _extract_player_positions_helper(
+def _extract_team_metadata_helper(
     side_data: TeamFrameInfo,
-) -> list[list[float]] | list:
+) -> FrameTeamMetadataDict:
     """Helper function to parse player locations in given side_data.
 
     Args:
-        side_data (TeamFrameInfo)       : Dict holding data for sides' players.
-                                          Expects frame['t'] or frame['ct'] to
-                                          be passed in.
-
-    Returns: List of player locations
+        side_data (TeamFrameInfo): Object with metadata for side's players.
 
-    Raises:
-        ValueError: If side_data not in TeamFrameInfo format
+    Returns: FrameTeamMetadataDict with metadata on team's players
     """
-    if "players" not in side_data:
-        raise ValueError("side_data arg not in expected TeamFrameInfo format.")
-
-    coords = ["x", "y", "z"]
-    alive_players: list[list[float]] | list = []
+    coords = ("x", "y", "z")
+    alive_players: list[list[float]] | list = [
+        [player[dim] for dim in coords]
+        for player in side_data["players"] or []
+        if player["isAlive"]
+    ]
 
-    for player in side_data["players"] or []:
-        # Players that are not alive are ignored
-        if player["isAlive"]:
-            alive_players.append([player[dim] for dim in coords])
-    return alive_players
+    return FrameTeamMetadataDict({"alive_player_locations": alive_players})
 
 
-def extract_player_positions(
+def extract_teams_metadata(
     frame: GameFrame,
-) -> dict[str, dict[str, list[list[float]]]]:
+) -> FrameTeamMetadataObject:
     """Parse frame data for alive player locations for both sides.
 
     Args:
-        frame (GameFrame)  : Dictionary in the form of an awpy frame
-                             containing relevant data for both sides
+        frame (GameFrame): Dictionary in the form of an awpy frame
+                           containing relevant data for both sides
 
```

Comment:
These should just be indented one normal level from the same as `frame`

So
```
 """Parse frame data for alive player locations for both sides.
    Args:
        frame (GameFrame): Dictionary in the form of an awpy frame
            containing relevant data for both sides
```

Same also further down.

Reply:
Addressed in next commit to branch. Also cleaned up docstrings I saw that were incorrect.